docs/src/main/asciidoc/security-jwt.adoc

```diff
@@ -794,6 +794,19 @@ public class TestJWTCallerPrincipalFactory extends JWTCallerPrincipalFactory {
 }
 ----
 
+=== Blocking calls
+
+`quarkus-smallrye-jwt` extension uses link:https://github.com/smallrye/smallrye-jwt[SmallRye JWT] library which is currently not reactive.
+
+What it means from the perspective of `quarkus-smallrye-jwt` which operates as part of the reactive Quarkus security architecture, is that an IO thread entering the link:https://github.com/smallrye/smallrye-jwt[SmallRye JWT] verification or decryption code might block in one of the following cases:
+
+* Default key resolver refreshes `JsonWebKey` set containing the keys which involves a remote call to the OIDC endpoint
+* Custom key resolver such as `AWS Application Load Balancer` (`ALB`) key resolver, resolves the keys against the AWS ALB key endpoint using the current token's key identifier header value
+
+In such cases, if the connections are slow, for example, it may take more than 3 seconds to get a response from the key endpoint, the current event loop thread will most likely block.
+
+To prevent it, set `quarkus.smallrye-jwt.blocking-authentication=true`.
```

Comment:
@AdamBien FYI - it may not be necessary, please check. Quarkus is super fast either way, but `smallrye-jwt-reactive` is on the map, just not sure yet when it will happen, but it will eventually

Reply:
I think the key retrieval / resolution could be a problem:

```
(vertx-blocked-thread-checker) Thread Thread[vert.x-eventloop-thread-0,5,main] has been blocked for 2557 ms, time limit is 2000 ms: io.vertx.core.VertxException: Thread blocked
	at java.base@17.0.6/java.lang.Thread.sleep(Native Method)
	at org.jose4j.http.Get.get(Get.java:111)
	at io.smallrye.jwt.auth.principal.AwsAlbKeyResolver.retrieveKey(AwsAlbKeyResolver.java:62)
	at io.smallrye.jwt.auth.principal.AwsAlbKeyResolver.resolveKey(AwsAlbKeyResolver.java:49)
	at org.jose4j.jwt.consumer.JwtConsumer.processContext(JwtConsumer.java:213)
	at org.jose4j.jwt.consumer.JwtConsumer.process(JwtConsumer.java:426)
	at io.smallrye.jwt.auth.principal.DefaultJWTTokenParser.parseClaims(DefaultJWTTokenParser.java:157)
	at io.smallrye.jwt.auth.principal.DefaultJWTTokenParser.parse(DefaultJWTTokenParser.java:60)
	at io.smallrye.jwt.auth.principal.DefaultJWTCallerPrincipalFactory.parse(DefaultJWTCallerPrincipalFactory.java:31)
	at io.smallrye.jwt.auth.principal.DefaultJWTParser.parse(DefaultJWTParser.java:71)
	at io.smallrye.jwt.auth.principal.DefaultJWTParser_ClientProxy.parse(Unknown Source)
	at io.quarkus.smallrye.jwt.runtime.auth.MpJwtValidator.createSecurityIdentity(MpJwtValidator.java:76)
```
The configuration:

`quarkus.smallrye-jwt.blocking-authentication=true`

solved the problem.